pkg/handlers/primeapi/payloads/model_to_payload.go

```
@@ -560,6 +565,61 @@ func Reweigh(reweigh *models.Reweigh) *primemessages.Reweigh {
 	return payload
 }
 
+// ExcessWeightRecord returns the fields on the move related to excess weights,
+// and returns the uploaded document set as the ExcessWeightUpload on the move.
+func ExcessWeightRecord(storer storage.FileStorer, move *models.Move) *primemessages.ExcessWeightRecord {
+	if move == nil || move.ID == uuid.Nil {
+		return nil
+	}
+
+	payload := &primemessages.ExcessWeightRecord{
+		MoveID:                         handlers.FmtUUIDPtr(&move.ID),
+		MoveExcessWeightQualifiedAt:    handlers.FmtDateTimePtr(move.ExcessWeightQualifiedAt),
+		MoveExcessWeightAcknowledgedAt: handlers.FmtDateTimePtr(move.ExcessWeightAcknowledgedAt),
+	}
+
+	upload := Upload(storer, move.ExcessWeightUpload)
+	if upload != nil {
+		payload.Upload = *upload
+	}
+
+	return payload
+}
+
+// Upload returns the data for an uploaded file.
+func Upload(storer storage.FileStorer, upload *models.Upload) *primemessages.Upload {
+	if upload == nil || upload.ID == uuid.Nil {
+		return nil
+	}
+
+	payload := &primemessages.Upload{
+		ID:          strfmt.UUID(upload.ID.String()),
+		Bytes:       &upload.Bytes,
+		ContentType: &upload.ContentType,
+		Filename:    &upload.Filename,
+		CreatedAt:   strfmt.DateTime(upload.CreatedAt),
+		UpdatedAt:   strfmt.DateTime(upload.UpdatedAt),
+	}
+
+	url, err := storer.PresignedURL(upload.StorageKey, upload.ContentType)
```

this error has been swallowed. I think we need a log line or catch of some form.

---

yeah, I think I'm going to add logging. It's a little weird because at this point, throwing errors is _kind_ of pointless because the update has already happened. I'd rather the user who called see that info is missing in the response. _But_ it's such a strange case we would want to see extra info in the logs, I think.

---

Sounds good!